packages/compass-connections-navigation/src/base-navigation-item.tsx

```diff
@@ -93,14 +97,17 @@ export const NavigationBaseItem: React.FC<NavigationBaseItemProps> = ({
   isExpandDisabled,
   isExpanded,
   isFocused,
+  hasDefaultAction,
   onExpand,
   children,
 }) => {
   const [hoverProps, isHovered] = useHoverState();
   return (
     <div
       data-testid="base-navigation-item"
-      className={itemContainerStyles}
+      className={cx(itemContainerStyles, {
+        [itemContainerWithActionStyles]: hasDefaultAction,
```

Comment:
Ideally this would be co-located with or derived from the implementation in `onDefaultAction` in `connections-navigation-tree.tsx`, but that is not trivial because of the virtualization.